File: Code/Components/DanpheEMR.ServerModel/MasterModels/ICD10Groups/MorbidityReportingGroupVM.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DanpheEMR.ServerModel
{
    public class MorbidityReportingGroupVM
    {
        public int ReportingGroupId { get; set; }
        public string ReportingGroupName { get; set; }
        public string GroupCode { get; set; }
        public int SerialNumber { get; set; }

        public List<MorbidityDiseaseGroupVM> DiseasesGroup = new List<MorbidityDiseaseGroupVM>();
    }


    public class MorbidityDiseaseGroupVM
    {
        public int SerialNumber { get; set; }
        public string ICDCode { get; set; }
        public string DiseaseGroupName { get; set; }
        public int NumberOfMale { get; set; }
        public int NumberOfFemale { get; set; }
        public int NumberOfOtherGender { get; set; }
    }
}

File: Code/Websites/DanpheEMR/wwwroot/DanpheApp/src/app/medical-records/shared/MorbidityReportingGroupVM.ts

export class MorbidityReportingGroupVM {
    public ReportingGroupId: string;
    public ReportingGroupName: string;
    public IcdVersion: string;
    public DiseasesGroup: string | MorbidityDiseaseGroupVM[];
}


export class MorbidityDiseaseGroupVM {
    public SerialNumber: number;
    public ICDCode: string;
    public DiseaseGroupName: string;
    public NumberOfMale: number;
    public NumberOfFemale: number;
    public NumberOfOtherGender: number;
}

export class EMERMorbidityReportingGroupVM {
    public ReportingGroupId: string;
    public ReportingGroupName: string;
    public IcdVersion: string;
    public DiseasesGroup: string | EMERMorbidityDiseaseGroupVM[];
}

export class EMERMorbidityDiseaseGroupVM {
    public SerialNumber: number;
    public ICDCode: string;
    public DiseaseGroupName: string;
    public NumberOfMale: number;
    public NumberOfFemale: number;
    public NumberOfOtherGender: number;
}
